darts/timeseries.py

```diff
@@ -560,10 +578,17 @@ def plot(self,
         kwargs
             some keyword arguments for the `plot()` method
         """
-
+        raise_if(self.width > 15, "Current TimeSeries instance contains too many components to plot.", logger)
         fig = (plt.figure() if new_plot else (kwargs['figure'] if 'figure' in kwargs else plt.gcf()))
         kwargs['figure'] = fig
-        self.pd_series().plot(*args, **kwargs)
+        if 'label' in kwargs:
+            label = kwargs['label']
+        for i in range(self.width):
+            if i > 0:
+                kwargs['figure'] = plt.gcf()
```

Comment:
maybe check that `figure` is not already in the kwargs (if the user specified a figure already)?